test/unit_tests/graphql/test_product.py

```diff
@@ -0,0 +1,49 @@
+import json
+from http import HTTPStatus
+from typing import Union
+
+from httpx import Response
+
+
+def get_product_query(
+    first: int = 10,
+    after: int = 0,
+    filter_by: Union[list[str], None] = None,
+    sort_by: Union[list[dict[str, str]], None] = None,
+) -> str:
+    query = """
+query ProductQuery($first: Int!, $after: Int!, $sortBy: [GraphqlSort!], $filterBy: [GraphqlFilter!]) {
+  products(first: $first, after: $after, sortBy: $sortBy, filterBy: $filterBy) {
+      name
+      productType
+      tag
+      status
+      productId
+      description
+      createdAt
+    }
+}
+    """
+    return json.dumps(
+        {
+            "operationName": "ProductQuery",
+            "query": query,
+            "variables": {
+                "first": first,
+                "after": after,
+                "sortBy": sort_by if sort_by else [],
+                "filterBy": filter_by if filter_by else [],
+            },
+        }
+    )
+
+
+def test_product_query(test_client, test_product_list_nested):
```

Comment:
add tests with range, sort and filter, you can look at `test_processes`

Reply:
Have a few more tests to add still, but made progress

Reply:
yea, could you add a sorting test, further it looks good

Reply:
Added! Hopefully these tests are sufficient for now